vite.config.js

```diff
@@ -63,14 +63,39 @@ export default defineConfig({
   define: {
     'process.env': {},
   },
-  plugins: [react()],
+  plugins: [
+    react(),
+    {
+      /**
+        * Middleware to rewrite HTML URLs to point to the deep path
+       */
+      configureServer(server) {
+        server.middlewares.use((req, res, next) => {
+          // Strip query parameters
+          const url = req.url.split('?')[0];
+          if (url.endsWith('.html') && !url.startsWith('/__tests__/integration/mirador/')) {
+            // rewrite root requests like /index.html to deep path
+            const newUrl = `/__tests__/integration/mirador${url}`;
+            req.url = newUrl;
+          }
```

Comment:
One more thing:

```suggestion
          } else if (url === '/') {
            req.url = '/__tests__/integration/mirador/index.html';
          }
```
Not strictly necessary as the Browser automatically opens with the correct URL but this would allow you to type in http://localhost:4444 directly and still being served something.

Reply:
Nice, thank you!